### ISSUE TITLE
Require InstancePrincipal for terraform-provider-oci for resource discovery

### ISSUE THREAD

**Issue by ivan20201** [terraform-providers/terraform-provider-oci#915]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

this module does not support instanceprinicpal authentication. It would be nice if this can be included in future release as in our environment user credential authentication not allowed, only instanceprincipal authentication alllowed.

### New or Affected Resource(s)

terraform-provider-oci

### Potential Terraform Configuration

```hcl
# Copy-paste any Terraform configurations for how the requested feature may be used. 
```
 ./terraform-provider-oci_v3.50.0_x4 -command=export -compartment_name=ocid1.tenancy.oc1..dfsddfsdsdfsdadsfadfdsa53543iboadfds ./export
2019/11/01 20:43:30 [INFO] terraform-provider-oci 3.50.0
can not create client, bad configuration: did not find a proper configuration for user

**Comment by hyder** [comment]
As long as your compute is in the instance_principal dynamic group and you've set the OCI_CLI_AUTH=instance_principal, then you should be able to run this command on a compute instance

**Comment by parrneet** [comment]
@ivan20201 InstancePrincipal auth is released for resource discovery with [v3.70.0](https://github.com/terraform-providers/terraform-provider-oci/releases/tag/v3.70.0)
Feel free to reopen the ticket if you face any issues.
